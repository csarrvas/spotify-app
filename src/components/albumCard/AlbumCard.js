import React from 'react';
import './albumCard.scss';
import { Link } from 'react-router-dom';

const AlbumCard = ({ album }) => {
  console.log(album);

  const displayArtistName = artists => {
    if (artists.length === 0) {
      return <p>No artist found for this album</p>
    } else {
      return <p>{(artists.reduce((all, artist) => all + artist.name + ', ', '')).slice(0, -2)}</p>
    }
  }

  return (
    <div className="album-card">
      <div>
        <p className="title">{album.artists.length > 1 ? 'Artists' : 'Artist'}</p>
        {displayArtistName(album.artists)}
      </div>
      <div>
        <p className="title">Go to album</p>
        <a target="_blank" rel="noopener noreferrer" href={album.external_urls.spotify}>Listen on spotify</a>
      </div>
      <figure>
        <img alt={`album ${album.name}`} src={album.images['1'].url}/>
      </figure>
      <div>
        <p>{album.name}</p>
      </div>
      <div>
        <Link to={`/albums/${album.id}`}><button>Album detail</button></Link>
      </div>
    </div>
  );
}
 
export default AlbumCard;