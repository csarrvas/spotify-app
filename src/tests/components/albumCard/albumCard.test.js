import React from 'react';
import { shallow } from 'enzyme';
import AlbumCard from '../../../components/albumCard/AlbumCard';
import { album } from '../../fixtures/data';

test('should render AlbumCard correctly', () => {
  const wrapper = shallow(<AlbumCard album={album}/>);
  expect(wrapper).toMatchSnapshot();
});