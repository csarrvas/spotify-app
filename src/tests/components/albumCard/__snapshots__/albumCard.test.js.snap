// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AlbumCard correctly 1`] = `
<div
  className="album-card"
>
  <div>
    <p>
      50 Cent
    </p>
  </div>
  <figure>
    <img
      alt="album Get Rich Or Die Tryin'"
      src="https://i.scdn.co/image/ab67616d00001e02d843fabb75fef14010e30cae"
    />
  </figure>
  <div>
    <p>
      Get Rich Or Die Tryin'
    </p>
  </div>
  <div>
    <a
      href="https://open.spotify.com/album/5G5rgQHzdQnw32SI0WjIo5"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fas fa-play-circle"
      >
         Go to this album on spotify
      </i>
    </a>
  </div>
  <div>
    <Link
      to="/albums/5G5rgQHzdQnw32SI0WjIo5"
    >
      <button>
        Album detail
      </button>
    </Link>
  </div>
</div>
`;
